tests/utils.py

```diff
@@ -649,7 +649,7 @@ def run(self, result=None):
         for test_name in dir(self):
             if test_name.startswith('test_'):
                 expect_failure = hasattr(self, 'not_implemented') and test_name in self.not_implemented
-                getattr(self, test_name).__dict__['__unittest_expecting_failure__'] = expect_failure
+                getattr(self, test_name).__dict__.setdefault('__unittest_expecting_failure__', expect_failure)
```

Comment:
The failure was the result of copy-pasting the solution from other places (Binary..., Builtin...), while the Unary case sometimes has no `not_implemented`. I think `setdefault` was the least invasive solution, because on other places there is no `expect_failure` intermediary.

Of course, could also write

    expect_failure = getattr(getattr(self, test_name), '__unittest_expecting_failure__', False) or test_name in getattr(self, 'not_implemented', [])
    getattr(self, test_name).__dict__['__unittest_expecting_failure__'] = expect_failure

which looks a 'tad' convoluted (especially the `expect_failure`).

Anyway, after re-thinking it a bit, I think it might be better to replace the custom `run` re-implementations with an extra Mixin class `NotImplementedToExpectedFailure` or something. That way it is more reasonable to get away with a bit more messy code, as it's really nicely moved away to its own class. However, in that case I'm not quite sure if it would be better to have all separate test-cases inherit from the mixin, or to make `UnaryOperationTestCase` extend from the mixin.